### ISSUE TITLE
Anomaly: uncaught exception Evarsolve.IllTypedInstance(_, _, _)

### ISSUE THREAD

**Issue by anishathalye** [coq/coq#13266]
#### Description of the problem

Here's a minimized example:

```coq
Inductive proc : Type -> Type :=
| Tick : proc unit
.

Inductive exec :
  forall T, proc T -> T -> Prop :=
| ExecTick :
    exec _ (Tick) tt
.

Lemma foo :
  exec _ Tick tt ->
  True.
Proof.
  intros H. 
  remember Tick as p.
  induction H.
```

That last induction step causes:

```
Error: Failed to instantiate evar: Anomaly "Uncaught exception Evarsolve.IllTypedInstance(_, _, _)."
Please report at http://coq.inria.fr/bugs/.
```

#### Coq Version

```console
$ coqtop -v
The Coq Proof Assistant, version 8.12.0 (July 2020)
compiled on Jul 26 2020 6:44:21 with OCaml 4.09.0
$ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.14.6
BuildVersion:   18G1012
```

**Comment by SkySkimmer** [comment]
Possibly related to https://github.com/coq/coq/issues/5512

**Comment by herbelin** [comment]
Fixed in #13376. It now reports instead about the unability to find a well-typed elimination predicate. Thanks for telling!
